docs/lazy-adr/adr-006-row-propagation.md

```diff
@@ -0,0 +1,195 @@
+# ADR 006: Consensus Block Gossiping with Rows
+
+## Changelog
+* 24.06.2021 - Initial description
+* 07.07.2021 - More important datails were added
+
+## Context
+It's a long story of relations between Celestia, Tendermint, and consensus block gossiping. Celestia's team discussed
+multiple ideas, several ADRs were made, and nothing yet was finalized. This ADR is another attempt to bring valuable
+changes into block gossiping and hopefully successful.
+
+Currently, we inherit the following from Tendermint. Our codebase relies on the blocks Parts notion. Each Part is a
+piece of an entire serialized block. Those Parts are gossiped between nodes in consensus and committed with
+`PartSetHeader` containing a Merkle Root of the Parts. However, Parts gossiping wasn't designed for Celestia blocks.
+
+Celestia comes with a different block representation from Tendermint. It lays out Blocks as a table of data shares,
+where Rows or Columns can be and should be gossiped instead of Parts, keeping only one system-wide commitment to data.
+
+## Alternative Approaches
```

Comment:
Let's add other alternative approaches and merge this PR. Alternatives are:

 - get rid of the PartsHeader from BlockID without changing block propagation at all
 - change block propagation to fixed-sized chunks but based on the ODS instead of how Parts are built currently
 - send the block as whole (only works with smaller blocks)

Reply:
@liamsi, should you do a suggestion to add more alternative approaches, or are you ok if I just copy and commit that without giving you a cred in a commit history?

Reply:
I'm OK with that.